ml/src/main/scala/ai/h2o/sparkling/ml/models/H2OTargetEncoderModel.scala

```diff
@@ -31,16 +32,22 @@ import water.api.schemas3.KeyV3.FrameKeyV3
 
 import scala.collection.JavaConverters._
 
-class H2OTargetEncoderModel(override val uid: String, targetEncoderModel: H2OModel)
+class H2OTargetEncoderModel(override val uid: String, targetEncoderModel: Option[H2OModel])
   extends Model[H2OTargetEncoderModel]
   with H2OTargetEncoderBase
   with MLWritable
   with RestCommunication {
 
-  lazy val mojoModel: H2OTargetEncoderMOJOModel = {
-    val mojo = targetEncoderModel.downloadMojo()
-    val model = new H2OTargetEncoderMOJOModel()
-    copyValues(model).setMojo(mojo)
+  lazy val mojoModel: H2OTargetEncoderMOJOModel = targetEncoderModel match {
+    case Some(targetEncoderModel) =>
+      val mojo = targetEncoderModel.downloadMojo()
+      val model = new H2OTargetEncoderMOJOModel()
+      copyValues(model).setMojo(mojo)
+    case None =>
+      val model = new H2OTargetEncoderNopMOJOModel()
```

Comment:
Why do we need this special case? Can it be handled by "empty" instance of H2OTargetEncoderMOJOModel? This way the implementations can diverge.

Reply:
Ok, I will try to remove `H2OTargetEncoderNopMOJOModel`